packages/components/src/tab-panel/types.ts

```diff
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import type { ReactNode } from 'react';
+import type { MouseEvent, ReactNode } from 'react';
```

Comment:
Fixes a TS error in [tab-panel/index.tsx](https://github.com/WordPress/gutenberg/blob/e702dd475b9c8eba500f9fd2eea58c03bc7f728d/packages/components/src/tab-panel/index.tsx#L26):

```
packages/components/src/tab-panel/index.tsx:26:3 - error TS2322: Type '{ children: ReactNode; label?: string | undefined; onClick: (event: MouseEvent) => void; showTooltip?: boolean | undefined; className?: string | undefined; icon?: IconType<...> | undefined; ... 5 more ...; __experimentalIsFocusable: true; }' is not assignable to type 'Omit<Pick<DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, "key" | keyof ButtonHTMLAttributes<...>> | Pick<...>, "as" | keyof ButtonProps>'.
  Types of property 'onClick' are incompatible.
    Type '(event: MouseEvent) => void' is not assignable to type 'MouseEventHandler<HTMLButtonElement> | MouseEventHandler<HTMLAnchorElement> | undefined'.
      Type '(event: MouseEvent) => void' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
        Types of parameters 'event' and 'event' are incompatible.
          Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is missing the following properties from type 'MouseEvent': offsetX, offsetY, x, y, and 14 more.

26  <Button
```

This uses `React.MouseEvent` instead of `MouseEvent`